packages/workbox-google-analytics/src/index.js

```diff
@@ -56,83 +49,122 @@ import replayQueuedRequests from './lib/replay-queued-requests.js';
-          fetch(request).then((response) => {
-            if (previousHitFailed) {
-              replayQueuedRequests(config);
-            }
-            previousHitFailed = false;
-            return response;
-          }, (error) => {
-            logHelper.log('Enqueuing failed request...');
-            previousHitFailed = true;
-            return enqueueRequest(clonedRequest).then(() => Response.error());
-          })
-        );
-      } else if (url.pathname === constants.URL.ANALYTICS_JS_PATH) {
-        // If this is a request for the Google Analytics JavaScript library,
-        // use the network first, falling back to the previously cached copy.
-        event.respondWith(
-          caches.open(constants.CACHE_NAME).then((cache) => {
-            return fetch(request).then((response) => {
-              return cache.put(request, response.clone()).then(() => response);
-            }).catch((error) => {
-              logHelper.error(error);
-              return cache.match(request);
-            });
-          })
-        );
+const createRequestWillDequeueCallback = (config) => {
+  return (reqData) => {
+    const {request} = reqData;
+    const hitTime = reqData.metadata.creationTimestamp;
+    const queueTime = Date.now() - hitTime;
+    const params = request.body ? new URLSearchParams(request.body) :
+        new URL(request.url).searchParams;
+
+    // Set the qt param prior to apply the hitFilter or parameterOverrides.
+    params.set('qt', queueTime);
+
+    if (config.parameterOverrides) {
+      for (const param of Object.keys(config.parameterOverrides)) {
+        const value = config.parameterOverrides[param];
+        params.set(param, value);
       }
     }
+
+    if (typeof config.hitFilter == 'function') {
+      config.hitFilter.call(null, params);
+    }
+
+    request.body = params.toString();
+    request.method = 'POST';
+    request.mode = 'cors';
+    request.credentials = 'omit';
+    request.headers = '[["Content-Type", "text/plain"]]';
+    request.url = `https://${constants.URL.HOST}${constants.URL.COLLECT_PATH}`;
+  };
+};
+
+/**
+ * Creates GET and POST routes to catch failed Measurement Protocol hits.
+ * @param {Function} requestWillDequeueCallback The callback used by
+ *     background sync when requests are dequeued and retried.
+ * @return {Array<Route>} The created routes.
+ */
+const createCollectRoutes = (requestWillDequeueCallback) => {
+  const bgSyncQueuePlugin = new QueuePlugin({
+    dbName: constants.IDB.NAME,
+    maxRetentionTime: constants.STOP_RETRYING_AFTER,
+    callbacks: {requestWillDequeue: requestWillDequeueCallback},
   });
 
-  replayQueuedRequests(config);
+  const match = ({url}) => url.hostname == constants.URL.HOST &&
+      url.pathname == constants.URL.COLLECT_PATH;
+
+  const handler = new NetworkOnly({
+    requestWrapper: new RequestWrapper({plugins: [bgSyncQueuePlugin]}),
+  });
+
+  return [
+    new Route({method: 'GET', match, handler}),
```

Comment:
Can't this be a single Router where the match function checks and matches the requests?

Reply:
That really sucks. Opened: https://github.com/GoogleChrome/workbox/issues/777